web3_multi_provider/util.py

```diff
@@ -23,3 +24,21 @@ def sanitize_poa_response(method: RPCEndpoint, response: RPCResponse) -> None:
             except ExtraDataLengthError:
                 logger.debug({"msg": "PoA blockchain cleanup response."})
                 response["result"] = extradata_to_poa_cleanup(response["result"])
+
+
+def normalize_provider(uri: str) -> str:
+    """
+    If uri is an IP address returns as is.
+    If uri is a dns address returns two highest domains.
+    """
+    uri = re.sub(r'^https?://', '', uri.lower())
+
+    ip_match = re.match(r'^(\d{1,3}\.){3}\d{1,3}(:\d+)?$', uri)
+    if ip_match:
+        return uri
+
+    hostname = uri.split('/')[0]
+    parts = hostname.split('.')
+    if len(parts) >= 2:
```

Comment:
What if domain is `.co.uk`?
I propose do not cover this edge case, or use some lib